nebari/keycloak.py

```diff
@@ -110,3 +112,76 @@ def get_keycloak_admin_from_config(config):
         raise ValueError(f"Failed to connect to Keycloak server: {e}")
 
     return keycloak_admin
+
+
+def keycloak_rest_api_call(config=None, request: str = None):
+    """Communicate directly with the Keycloak REST API by passing it a request"""
+
+    config = load_yaml(config)
+
+    keycloak_server_url = os.environ.get(
+        "KEYCLOAK_SERVER_URL", f"https://{config['domain']}/auth"
+    )
+    keycloak_admin_username = os.environ.get("KEYCLOAK_ADMIN_USERNAME", "root")
+    keycloak_admin_password = os.environ.get(
+        "KEYCLOAK_ADMIN_PASSWORD",
+        config.get("security", {}).get("keycloak", {}).get("initial_root_password", ""),
+    )
+
+    try:
+        # Get `token` to interact with Keycloak Admin
+        url = keycloak_server_url + "realms/master/protocol/openid-connect/token"
+        headers = {
+            "Content-Type": "application/x-www-form-urlencoded",
+        }
+        data = {
+            "username": keycloak_admin_username,
+            "password": keycloak_admin_password,
+            "grant_type": "password",
+            "client_id": "admin-cli",
+        }
+
+        response = requests.post(
+            url=url,
+            headers=headers,
+            data=data,
+        )
+
+        if response.status_code == 200:
+            token = json.loads(response.content.decode())["access_token"]
+        else:
+            raise ValueError(
+                f"Unable to retrieve Keycloak API token. Status code: {response.status_code}"
+            )
+
+        # Send request to Keycloak REST API
+        method, endpoint = request.split()
+        url = keycloak_server_url + "admin/realms" + endpoint
```

Comment:
```suggestion
        url = keycloak_server_url + "/admin/realms" + endpoint
```
currently: `https://github-actions.nebari.dev/authadmin/realms/nebari/users`